refactor [#449] 네이밍 컨벤션 통일 및 코드리뷰 사항 반영

junggyo1020 opened the pull request:
<!-- 제목양식을 지켜주세요! Prefix [#이슈번호] {PR 설명} -->
<!-- PR 작성 후 우측에 Development에서 이슈 찾아서 연동하면 merge될때 이슈도 close됩니다 -->
<!-- Reviewer, Assignees, Label 붙이기 --> 
## 🔗 Issue Number
<!-- #{본인 이슈 번호} 치면 알아서 이슈 게시판 링크 걸려요 -->
closes #449 

## 📌  To-do
<!-- 본인이 한 업무를 체크리스트로 작성해주세요 -->

- [x] userId를 nullable 하지 않도록 수정
- [x] 메서드의 책임 분리
- [x] SetlistMusics 테이블의 trackId -> musicId로 컬럼명 변경
- [x] 기존에 사용되던 셋리스트 음악 관련 DTO의 musicId 필드명 변경 -> setlistMusicId
- [x] 셋리스트 관련 모든 API 테스트
- [x] API 명세서에 모든 수정사항 반영
- [x] 배포 전 API 수정사항 Slack에 공지


## ✨Description 
<!-- 본인이 한 작업을 설명해주세요 -->

### 코드리뷰 사항은 일부만 반영한 상황이고, 계속해서 반영해 별도의 PR로 올릴 예정입니다.

### userId를 nullable 하지 않도록 수정
Controller에서 Authentication 을 통해 전달받는 userId는 셋리스트 관련 API에서는 꼭 필요하므로 `nullable`하지 않도록 어노테이션을 수정했습니다!
```java
// 변경 전
@UserId(required = false) Long userId

//변경 후
@UserId Long userId
```
### 메서드의 책임 분리
기존의 updateMusicOrder의 경우 메서드의 길이가 15줄 이상이 넘어가면서 책임이 명확히 분리되어 있지 않아, 메서드를 책임에 맞게 분리하여 SRP를 준수하고자 하였습니다!
```
//변경 전
updateMusicOrder() -> redisKey 유효성 검증 및 역직렬화, Map 형식으로 변환, 순서 변경

//변경 후
updateMusicOrder() 
-> getRedisMusicList() : redisKey 유효성 검증 및 역직렬화
-> toMusicMap() : Map 형식으로 변환
-> swapOrder() : 순서 변경
```
### SetlistMusics 테이블의 trackId -> `musicId`로 컬럼명 변경
- + 기존에 사용되던 셋리스트 음악 관련 DTO의 musicId 필드명 변경 -> `setlistMusicId`
<img width="455" alt="image" src="https://github.com/user-attachments/assets/68c45e2d-5249-4280-b3f4-f831394b4910" />
<img width="469" alt="image" src="https://github.com/user-attachments/assets/555fc295-2941-4698-93cc-3d30b987e8d3" />

### 셋리스트 관련 모든 API 테스트 `완료`
### API 명세서에 모든 수정사항 반영 `완료`
### 배포 전 API 수정사항 Slack에 공지 `완료`
- https://weareconfeti.slack.com/archives/C08K7RMK1FS/p1748851625662369

junggyo1020 committed:
chore [#449] Optional 삭제

junggyo1020 committed:
chore [#449] Optional 삭제

junggyo1020 committed:
refactor [#449] 셋리스트 항목 순서 변경 비즈니스 로직 메서드를 통한 책임 분리

junggyo1020 committed:
refactor [#449] 셋리스트 항목 순서 변경 비즈니스 로직 메서드를 통한 책임 분리

junggyo1020 committed:
chore [#449] 테스트 코드 삭제

junggyo1020 committed:
chore [#449] 테스트 코드 삭제

junggyo1020 committed:
chore [#449] DTO 컬럼명 변경

- musicId -> setlistMusicId
- trackId -> musicId

junggyo1020 committed:
chore [#449] DTO 컬럼명 변경

- musicId -> setlistMusicId
- trackId -> musicId

junggyo1020 committed:
chore [#449] 컬럼명 변경

- trackId -> musicId

junggyo1020 committed:
chore [#449] 컬럼명 변경

- trackId -> musicId

junggyo1020 committed:
refactor [#449] 컬럼명 변경에 따른 파라미터 수정

- trackId -> musicId

junggyo1020 committed:
refactor [#449] 컬럼명 변경에 따른 파라미터 수정

- trackId -> musicId

junggyo1020 committed:
refactor [#449] 메서드명 통일

- create -> from

junggyo1020 committed:
refactor [#449] 메서드명 통일

- create -> from

ch1hyun committed:
setting [#436] Spring Rest Docs + Open API 명세 생성 및 Swagger 연동

* feat [#436] 테스트 기본 데이터 작성

* feat [#436] 엘라스틱 서치 인덱스, 매핑 구성 파일 작성

* feat [#436] Test Container 사용을 위한 세팅

* feat [#436] 테스트용 Elastic Search 구성 정보 작성

* feat [#436] 테스트 기본 추상 클래스 작성

* feat [#436] Test Container, Spring Rest Docs 및 Open API 자동 생성을 위한 라이브러리 추가

* feat [#436] 인덱스 구성 정보 파일 연결

* refactor [#436] 기본 값 수정

* fix [#436] 검색어로 아티스트를 검색했을 때도 기록에 수집되도록 추가

* feat [#436] 전역 테스트 컨테이너 생성 및 관리로 수정

* refactor [#436] 하이버네이트 캐시 오류로 매번 재생성되도록 수정

* feat [#436] 각 테스트마다 환경의 멱등성을 보장하기 위해 데이터 초기화 로직 추가

* feat [#436] 예제 테스트 작성

- Artist Search
- User Favorite
- User Timetable

* feat [#436] Redis Test Container 추가

* refactor [#436] 로깅 삭제

* refactor [#436] 폴더 경로 수정 및 역할에 맞게 클래스 분리

* feat [#436] Swagger UI 연동

* feat [#436] Swagger UI 의존성 추가

* refactor [#436] openapi3.yml 노출 제거

* style [#436] 공백 추가

* build [#436] CI/CD Work Flow에 application-test.yml 구성 추가

* RestDocs 적용으로 사용되지 않는 테스트 코드 주석처리

이후 테스트 코드 작성 시 참고할 예정이라 삭제가 아닌 주석처리 하였습니다.

* chore [#436] 추후 사용해야 할 테스트 코드 주석처리

* chore [#436] 추후 사용해야 할 테스트 코드 주석처리

* chore [#436] 사용하지 않는 테스트 코드 삭제

* build [#436] CI/CD 워크플로우에 테스트 컨테이너 reuse 설정 추가

* build [#436] CI/CD 워크플로우에 오류 리포트 확인 추가

* build [#436] CI/CD 워크플로우 빌드 디버깅 추가

* build [#436] CI/CD 워크플로우 도커 허용

* build [#436] CI/CD 워크플로우 수정

* refactor [#436] 사용되지 않는 테스트 제거

* refactor [#436] CI/CD 필요없는 구성 정보 제거

* refactor [#436] CD Test Container reuse 로컬 우선순위 강화

* refactor [#452] profileUrl(String) -> profileFile(MultipartFile) 변경 (#453)

* refactor [#436] 유저 디스코드 알림 연동 감시를 위한 로그 추가

* feat [#436] 테스트 기본 데이터 작성

* feat [#436] 엘라스틱 서치 인덱스, 매핑 구성 파일 작성

* feat [#436] Test Container 사용을 위한 세팅

* feat [#436] 테스트용 Elastic Search 구성 정보 작성

* feat [#436] 테스트 기본 추상 클래스 작성

* feat [#436] Test Container, Spring Rest Docs 및 Open API 자동 생성을 위한 라이브러리 추가

* feat [#436] 인덱스 구성 정보 파일 연결

* refactor [#436] 기본 값 수정

* fix [#436] 검색어로 아티스트를 검색했을 때도 기록에 수집되도록 추가

* feat [#436] 전역 테스트 컨테이너 생성 및 관리로 수정

* refactor [#436] 하이버네이트 캐시 오류로 매번 재생성되도록 수정

* feat [#436] 각 테스트마다 환경의 멱등성을 보장하기 위해 데이터 초기화 로직 추가

* feat [#436] 예제 테스트 작성

- Artist Search
- User Favorite
- User Timetable

* feat [#436] Redis Test Container 추가

* refactor [#436] 로깅 삭제

* refactor [#436] 폴더 경로 수정 및 역할에 맞게 클래스 분리

* feat [#436] Swagger UI 연동

* feat [#436] Swagger UI 의존성 추가

* refactor [#436] openapi3.yml 노출 제거

* style [#436] 공백 추가

* build [#436] CI/CD Work Flow에 application-test.yml 구성 추가

* RestDocs 적용으로 사용되지 않는 테스트 코드 주석처리

이후 테스트 코드 작성 시 참고할 예정이라 삭제가 아닌 주석처리 하였습니다.

* chore [#436] 추후 사용해야 할 테스트 코드 주석처리

* build [#436] CI/CD 워크플로우에 테스트 컨테이너 reuse 설정 추가

* build [#436] CI/CD 워크플로우에 오류 리포트 확인 추가

* build [#436] CI/CD 워크플로우 빌드 디버깅 추가

* build [#436] CI/CD 워크플로우 도커 허용

* build [#436] CI/CD 워크플로우 수정

* refactor [#436] 사용되지 않는 테스트 제거

* refactor [#436] CI/CD 필요없는 구성 정보 제거

* refactor [#436] CD Test Container reuse 로컬 우선순위 강화

* refactor [#436] 유저 디스코드 알림 연동 감시를 위한 로그 추가

---------

Co-authored-by: Willy <willy_1020@naver.com>
Co-authored-by: Ivoryeee <105477246+Ivoryeee@users.noreply.github.com>

ch1hyun committed:
feat [#446] 검색 API 통합

* refactor [#446] 클래스 명 변경

- SearchTermController -> SearchController

* feat [#446] 홈 검색 API 아티스트 아이디로 조회하는 기능 구현

* feat [#446] 홈 검색 API 아티스트 아이디로 조회하는 기능에서 검색어 저장

* feat [#446] 홈 검색 API 공연 아이디로 조회하는 기능 구현

* feat [#446] 검색어 분석 유틸 클래스 작성

* feat [#446] 홈 검색 API 검색어로 조회하는 기능 구현

* refactor [#446] 더 이상 사용되지 않는 검색 API 제거

* refactor [#452] profileUrl(String) -> profileFile(MultipartFile) 변경 (#453)

* refactor [#446] 검색 쿼리에 따른 분기 처리 리펙토링

* setting [#436] Spring Rest Docs + Open API 명세 생성 및 Swagger 연동

* feat [#436] 테스트 기본 데이터 작성

* feat [#436] 엘라스틱 서치 인덱스, 매핑 구성 파일 작성

* feat [#436] Test Container 사용을 위한 세팅

* feat [#436] 테스트용 Elastic Search 구성 정보 작성

* feat [#436] 테스트 기본 추상 클래스 작성

* feat [#436] Test Container, Spring Rest Docs 및 Open API 자동 생성을 위한 라이브러리 추가

* feat [#436] 인덱스 구성 정보 파일 연결

* refactor [#436] 기본 값 수정

* fix [#436] 검색어로 아티스트를 검색했을 때도 기록에 수집되도록 추가

* feat [#436] 전역 테스트 컨테이너 생성 및 관리로 수정

* refactor [#436] 하이버네이트 캐시 오류로 매번 재생성되도록 수정

* feat [#436] 각 테스트마다 환경의 멱등성을 보장하기 위해 데이터 초기화 로직 추가

* feat [#436] 예제 테스트 작성

- Artist Search
- User Favorite
- User Timetable

* feat [#436] Redis Test Container 추가

* refactor [#436] 로깅 삭제

* refactor [#436] 폴더 경로 수정 및 역할에 맞게 클래스 분리

* feat [#436] Swagger UI 연동

* feat [#436] Swagger UI 의존성 추가

* refactor [#436] openapi3.yml 노출 제거

* style [#436] 공백 추가

* build [#436] CI/CD Work Flow에 application-test.yml 구성 추가

* RestDocs 적용으로 사용되지 않는 테스트 코드 주석처리

이후 테스트 코드 작성 시 참고할 예정이라 삭제가 아닌 주석처리 하였습니다.

* chore [#436] 추후 사용해야 할 테스트 코드 주석처리

* chore [#436] 추후 사용해야 할 테스트 코드 주석처리

* chore [#436] 사용하지 않는 테스트 코드 삭제

* build [#436] CI/CD 워크플로우에 테스트 컨테이너 reuse 설정 추가

* build [#436] CI/CD 워크플로우에 오류 리포트 확인 추가

* build [#436] CI/CD 워크플로우 빌드 디버깅 추가

* build [#436] CI/CD 워크플로우 도커 허용

* build [#436] CI/CD 워크플로우 수정

* refactor [#436] 사용되지 않는 테스트 제거

* refactor [#436] CI/CD 필요없는 구성 정보 제거

* refactor [#436] CD Test Container reuse 로컬 우선순위 강화

* refactor [#452] profileUrl(String) -> profileFile(MultipartFile) 변경 (#453)

* refactor [#436] 유저 디스코드 알림 연동 감시를 위한 로그 추가

* feat [#436] 테스트 기본 데이터 작성

* feat [#436] 엘라스틱 서치 인덱스, 매핑 구성 파일 작성

* feat [#436] Test Container 사용을 위한 세팅

* feat [#436] 테스트용 Elastic Search 구성 정보 작성

* feat [#436] 테스트 기본 추상 클래스 작성

* feat [#436] Test Container, Spring Rest Docs 및 Open API 자동 생성을 위한 라이브러리 추가

* feat [#436] 인덱스 구성 정보 파일 연결

* refactor [#436] 기본 값 수정

* fix [#436] 검색어로 아티스트를 검색했을 때도 기록에 수집되도록 추가

* feat [#436] 전역 테스트 컨테이너 생성 및 관리로 수정

* refactor [#436] 하이버네이트 캐시 오류로 매번 재생성되도록 수정

* feat [#436] 각 테스트마다 환경의 멱등성을 보장하기 위해 데이터 초기화 로직 추가

* feat [#436] 예제 테스트 작성

- Artist Search
- User Favorite
- User Timetable

* feat [#436] Redis Test Container 추가

* refactor [#436] 로깅 삭제

* refactor [#436] 폴더 경로 수정 및 역할에 맞게 클래스 분리

* feat [#436] Swagger UI 연동

* feat [#436] Swagger UI 의존성 추가

* refactor [#436] openapi3.yml 노출 제거

* style [#436] 공백 추가

* build [#436] CI/CD Work Flow에 application-test.yml 구성 추가

* RestDocs 적용으로 사용되지 않는 테스트 코드 주석처리

이후 테스트 코드 작성 시 참고할 예정이라 삭제가 아닌 주석처리 하였습니다.

* chore [#436] 추후 사용해야 할 테스트 코드 주석처리

* build [#436] CI/CD 워크플로우에 테스트 컨테이너 reuse 설정 추가

* build [#436] CI/CD 워크플로우에 오류 리포트 확인 추가

* build [#436] CI/CD 워크플로우 빌드 디버깅 추가

* build [#436] CI/CD 워크플로우 도커 허용

* build [#436] CI/CD 워크플로우 수정

* refactor [#436] 사용되지 않는 테스트 제거

* refactor [#436] CI/CD 필요없는 구성 정보 제거

* refactor [#436] CD Test Container reuse 로컬 우선순위 강화

* refactor [#436] 유저 디스코드 알림 연동 감시를 위한 로그 추가

---------

Co-authored-by: Willy <willy_1020@naver.com>
Co-authored-by: Ivoryeee <105477246+Ivoryeee@users.noreply.github.com>

* refactor [#446] 클래스 명 변경

- SearchTermController -> SearchController

* feat [#446] 홈 검색 API 아티스트 아이디로 조회하는 기능 구현

* feat [#446] 홈 검색 API 아티스트 아이디로 조회하는 기능에서 검색어 저장

* feat [#446] 홈 검색 API 공연 아이디로 조회하는 기능 구현

* feat [#446] 검색어 분석 유틸 클래스 작성

* feat [#446] 홈 검색 API 검색어로 조회하는 기능 구현

* refactor [#446] 더 이상 사용되지 않는 검색 API 제거

* refactor [#446] 검색 쿼리에 따른 분기 처리 리펙토링

* refactor [#446] CI/CD Workflow에 Swagger 배포를 위한 폴더 생성 추가

* refactor [#446] 사용되지 않는 API 테스트 제거

---------

Co-authored-by: Ivoryeee <105477246+Ivoryeee@users.noreply.github.com>
Co-authored-by: Willy <willy_1020@naver.com>

Ivoryeee commented on src/main/java/org/sopt/confeti/domain/setlist/application/SetlistEditService.java:
```diff
@@ -169,4 +157,33 @@ public void cancelEdit(Long userId, Long setlistId) {
+        SetlistMusicEditDTO a = musicMap.get(requests.get(0).musicId());
+        SetlistMusicEditDTO b = musicMap.get(requests.get(1).musicId());
+
+        if (a != null && b != null) {
```
호옥시 여기는 `null` 사용을 유지하신 이유가 있을까요? 요 부분도 `Objects.nonNull()`로 수정할 수 있을 것 같아서 질문 드려요!

junggyo1020 replied:
제가 놓친 부분이네요! 이 부분도 반영해보겠습니다! 감사합니다 :)

Ivoryeee reviewed (approved):
LGTM! 수고 많으셨습니다 👍🏻 👍🏻

junggyo1020 committed:
chore [#449] 브랜치 충돌 해결

junggyo1020 committed:
chore [#449] 브랜치 충돌 해결

junggyo1020 committed:
chore [#449] Objects 메서드로 구체적인 의도 명시

junggyo1020 committed:
chore [#449] Objects 메서드로 구체적인 의도 명시

junggyo1020 committed:
chore [#449] type값 대문자로 넘어오도록 수정

junggyo1020 committed:
chore [#449] type값 대문자로 넘어오도록 수정

Ivoryeee committed:
feat [#454] 전체 공연 목록 조회 API 구현 (#456)

* feat [#454] 전체 공연 목록 조회 API 구현

* chore [#454] Swagger 설정을 위한 addResourceHandlers 추가

* setting [#436] Spring Rest Docs + Open API 명세 생성 및 Swagger 연동

* feat [#436] 테스트 기본 데이터 작성

* feat [#436] 엘라스틱 서치 인덱스, 매핑 구성 파일 작성

* feat [#436] Test Container 사용을 위한 세팅

* feat [#436] 테스트용 Elastic Search 구성 정보 작성

* feat [#436] 테스트 기본 추상 클래스 작성

* feat [#436] Test Container, Spring Rest Docs 및 Open API 자동 생성을 위한 라이브러리 추가

* feat [#436] 인덱스 구성 정보 파일 연결

* refactor [#436] 기본 값 수정

* fix [#436] 검색어로 아티스트를 검색했을 때도 기록에 수집되도록 추가

* feat [#436] 전역 테스트 컨테이너 생성 및 관리로 수정

* refactor [#436] 하이버네이트 캐시 오류로 매번 재생성되도록 수정

* feat [#436] 각 테스트마다 환경의 멱등성을 보장하기 위해 데이터 초기화 로직 추가

* feat [#436] 예제 테스트 작성

- Artist Search
- User Favorite
- User Timetable

* feat [#436] Redis Test Container 추가

* refactor [#436] 로깅 삭제

* refactor [#436] 폴더 경로 수정 및 역할에 맞게 클래스 분리

* feat [#436] Swagger UI 연동

* feat [#436] Swagger UI 의존성 추가

* refactor [#436] openapi3.yml 노출 제거

* style [#436] 공백 추가

* build [#436] CI/CD Work Flow에 application-test.yml 구성 추가

* RestDocs 적용으로 사용되지 않는 테스트 코드 주석처리

이후 테스트 코드 작성 시 참고할 예정이라 삭제가 아닌 주석처리 하였습니다.

* chore [#436] 추후 사용해야 할 테스트 코드 주석처리

* chore [#436] 추후 사용해야 할 테스트 코드 주석처리

* chore [#436] 사용하지 않는 테스트 코드 삭제

* build [#436] CI/CD 워크플로우에 테스트 컨테이너 reuse 설정 추가

* build [#436] CI/CD 워크플로우에 오류 리포트 확인 추가

* build [#436] CI/CD 워크플로우 빌드 디버깅 추가

* build [#436] CI/CD 워크플로우 도커 허용

* build [#436] CI/CD 워크플로우 수정

* refactor [#436] 사용되지 않는 테스트 제거

* refactor [#436] CI/CD 필요없는 구성 정보 제거

* refactor [#436] CD Test Container reuse 로컬 우선순위 강화

* refactor [#452] profileUrl(String) -> profileFile(MultipartFile) 변경 (#453)

* refactor [#436] 유저 디스코드 알림 연동 감시를 위한 로그 추가

* feat [#436] 테스트 기본 데이터 작성

* feat [#436] 엘라스틱 서치 인덱스, 매핑 구성 파일 작성

* feat [#436] Test Container 사용을 위한 세팅

* feat [#436] 테스트용 Elastic Search 구성 정보 작성

* feat [#436] 테스트 기본 추상 클래스 작성

* feat [#436] Test Container, Spring Rest Docs 및 Open API 자동 생성을 위한 라이브러리 추가

* feat [#436] 인덱스 구성 정보 파일 연결

* refactor [#436] 기본 값 수정

* fix [#436] 검색어로 아티스트를 검색했을 때도 기록에 수집되도록 추가

* feat [#436] 전역 테스트 컨테이너 생성 및 관리로 수정

* refactor [#436] 하이버네이트 캐시 오류로 매번 재생성되도록 수정

* feat [#436] 각 테스트마다 환경의 멱등성을 보장하기 위해 데이터 초기화 로직 추가

* feat [#436] 예제 테스트 작성

- Artist Search
- User Favorite
- User Timetable

* feat [#436] Redis Test Container 추가

* refactor [#436] 로깅 삭제

* refactor [#436] 폴더 경로 수정 및 역할에 맞게 클래스 분리

* feat [#436] Swagger UI 연동

* feat [#436] Swagger UI 의존성 추가

* refactor [#436] openapi3.yml 노출 제거

* style [#436] 공백 추가

* build [#436] CI/CD Work Flow에 application-test.yml 구성 추가

* RestDocs 적용으로 사용되지 않는 테스트 코드 주석처리

이후 테스트 코드 작성 시 참고할 예정이라 삭제가 아닌 주석처리 하였습니다.

* chore [#436] 추후 사용해야 할 테스트 코드 주석처리

* build [#436] CI/CD 워크플로우에 테스트 컨테이너 reuse 설정 추가

* build [#436] CI/CD 워크플로우에 오류 리포트 확인 추가

* build [#436] CI/CD 워크플로우 빌드 디버깅 추가

* build [#436] CI/CD 워크플로우 도커 허용

* build [#436] CI/CD 워크플로우 수정

* refactor [#436] 사용되지 않는 테스트 제거

* refactor [#436] CI/CD 필요없는 구성 정보 제거

* refactor [#436] CD Test Container reuse 로컬 우선순위 강화

* refactor [#436] 유저 디스코드 알림 연동 감시를 위한 로그 추가

---------

Co-authored-by: Willy <willy_1020@naver.com>
Co-authored-by: Ivoryeee <105477246+Ivoryeee@users.noreply.github.com>

* feat [#446] 검색 API 통합

* refactor [#446] 클래스 명 변경

- SearchTermController -> SearchController

* feat [#446] 홈 검색 API 아티스트 아이디로 조회하는 기능 구현

* feat [#446] 홈 검색 API 아티스트 아이디로 조회하는 기능에서 검색어 저장

* feat [#446] 홈 검색 API 공연 아이디로 조회하는 기능 구현

* feat [#446] 검색어 분석 유틸 클래스 작성

* feat [#446] 홈 검색 API 검색어로 조회하는 기능 구현

* refactor [#446] 더 이상 사용되지 않는 검색 API 제거

* refactor [#452] profileUrl(String) -> profileFile(MultipartFile) 변경 (#453)

* refactor [#446] 검색 쿼리에 따른 분기 처리 리펙토링

* setting [#436] Spring Rest Docs + Open API 명세 생성 및 Swagger 연동

* feat [#436] 테스트 기본 데이터 작성

* feat [#436] 엘라스틱 서치 인덱스, 매핑 구성 파일 작성

* feat [#436] Test Container 사용을 위한 세팅

* feat [#436] 테스트용 Elastic Search 구성 정보 작성

* feat [#436] 테스트 기본 추상 클래스 작성

* feat [#436] Test Container, Spring Rest Docs 및 Open API 자동 생성을 위한 라이브러리 추가

* feat [#436] 인덱스 구성 정보 파일 연결

* refactor [#436] 기본 값 수정

* fix [#436] 검색어로 아티스트를 검색했을 때도 기록에 수집되도록 추가

* feat [#436] 전역 테스트 컨테이너 생성 및 관리로 수정

* refactor [#436] 하이버네이트 캐시 오류로 매번 재생성되도록 수정

* feat [#436] 각 테스트마다 환경의 멱등성을 보장하기 위해 데이터 초기화 로직 추가

* feat [#436] 예제 테스트 작성

- Artist Search
- User Favorite
- User Timetable

* feat [#436] Redis Test Container 추가

* refactor [#436] 로깅 삭제

* refactor [#436] 폴더 경로 수정 및 역할에 맞게 클래스 분리

* feat [#436] Swagger UI 연동

* feat [#436] Swagger UI 의존성 추가

* refactor [#436] openapi3.yml 노출 제거

* style [#436] 공백 추가

* build [#436] CI/CD Work Flow에 application-test.yml 구성 추가

* RestDocs 적용으로 사용되지 않는 테스트 코드 주석처리

이후 테스트 코드 작성 시 참고할 예정이라 삭제가 아닌 주석처리 하였습니다.

* chore [#436] 추후 사용해야 할 테스트 코드 주석처리

* chore [#436] 추후 사용해야 할 테스트 코드 주석처리

* chore [#436] 사용하지 않는 테스트 코드 삭제

* build [#436] CI/CD 워크플로우에 테스트 컨테이너 reuse 설정 추가

* build [#436] CI/CD 워크플로우에 오류 리포트 확인 추가

* build [#436] CI/CD 워크플로우 빌드 디버깅 추가

* build [#436] CI/CD 워크플로우 도커 허용

* build [#436] CI/CD 워크플로우 수정

* refactor [#436] 사용되지 않는 테스트 제거

* refactor [#436] CI/CD 필요없는 구성 정보 제거

* refactor [#436] CD Test Container reuse 로컬 우선순위 강화

* refactor [#452] profileUrl(String) -> profileFile(MultipartFile) 변경 (#453)

* refactor [#436] 유저 디스코드 알림 연동 감시를 위한 로그 추가

* feat [#436] 테스트 기본 데이터 작성

* feat [#436] 엘라스틱 서치 인덱스, 매핑 구성 파일 작성

* feat [#436] Test Container 사용을 위한 세팅

* feat [#436] 테스트용 Elastic Search 구성 정보 작성

* feat [#436] 테스트 기본 추상 클래스 작성

* feat [#436] Test Container, Spring Rest Docs 및 Open API 자동 생성을 위한 라이브러리 추가

* feat [#436] 인덱스 구성 정보 파일 연결

* refactor [#436] 기본 값 수정

* fix [#436] 검색어로 아티스트를 검색했을 때도 기록에 수집되도록 추가

* feat [#436] 전역 테스트 컨테이너 생성 및 관리로 수정

* refactor [#436] 하이버네이트 캐시 오류로 매번 재생성되도록 수정

* feat [#436] 각 테스트마다 환경의 멱등성을 보장하기 위해 데이터 초기화 로직 추가

* feat [#436] 예제 테스트 작성

- Artist Search
- User Favorite
- User Timetable

* feat [#436] Redis Test Container 추가

* refactor [#436] 로깅 삭제

* refactor [#436] 폴더 경로 수정 및 역할에 맞게 클래스 분리

* feat [#436] Swagger UI 연동

* feat [#436] Swagger UI 의존성 추가

* refactor [#436] openapi3.yml 노출 제거

* style [#436] 공백 추가

* build [#436] CI/CD Work Flow에 application-test.yml 구성 추가

* RestDocs 적용으로 사용되지 않는 테스트 코드 주석처리

이후 테스트 코드 작성 시 참고할 예정이라 삭제가 아닌 주석처리 하였습니다.

* chore [#436] 추후 사용해야 할 테스트 코드 주석처리

* build [#436] CI/CD 워크플로우에 테스트 컨테이너 reuse 설정 추가

* build [#436] CI/CD 워크플로우에 오류 리포트 확인 추가

* build [#436] CI/CD 워크플로우 빌드 디버깅 추가

* build [#436] CI/CD 워크플로우 도커 허용

* build [#436] CI/CD 워크플로우 수정

* refactor [#436] 사용되지 않는 테스트 제거

* refactor [#436] CI/CD 필요없는 구성 정보 제거

* refactor [#436] CD Test Container reuse 로컬 우선순위 강화

* refactor [#436] 유저 디스코드 알림 연동 감시를 위한 로그 추가

---------

Co-authored-by: Willy <willy_1020@naver.com>
Co-authored-by: Ivoryeee <105477246+Ivoryeee@users.noreply.github.com>

* refactor [#446] 클래스 명 변경

- SearchTermController -> SearchController

* feat [#446] 홈 검색 API 아티스트 아이디로 조회하는 기능 구현

* feat [#446] 홈 검색 API 아티스트 아이디로 조회하는 기능에서 검색어 저장

* feat [#446] 홈 검색 API 공연 아이디로 조회하는 기능 구현

* feat [#446] 검색어 분석 유틸 클래스 작성

* feat [#446] 홈 검색 API 검색어로 조회하는 기능 구현

* refactor [#446] 더 이상 사용되지 않는 검색 API 제거

* refactor [#446] 검색 쿼리에 따른 분기 처리 리펙토링

* refactor [#446] CI/CD Workflow에 Swagger 배포를 위한 폴더 생성 추가

* refactor [#446] 사용되지 않는 API 테스트 제거

---------

Co-authored-by: Ivoryeee <105477246+Ivoryeee@users.noreply.github.com>
Co-authored-by: Willy <willy_1020@naver.com>

* chore [#454] type 대문자로 넘어오도록 변경

---------

Co-authored-by: chyun <och2510@gmail.com>
Co-authored-by: Willy <willy_1020@naver.com>

ch1hyun reviewed (approved):
LGTM!

junggyo1020 committed:
Merge remote-tracking branch 'origin/refactor/#449' into refactor/#449
